src/averageAges.js

```diff
@@ -38,6 +47,14 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  const searchedWomen = withChildren
+    ? people.filter(person => person.sex === 'f'
+      && people.some(children => children.mother === person.name))
+    : people.filter(person => person.sex === 'f');
+
+  const womenAgeArray = searchedWomen.map(person => person.died - person.born);
+
+  return womenAgeArray.reduce((a, b) => a + b) / womenAgeArray.length;
```

Comment:
2. [CODE STYLE] - If you move your shortened return statement in arrow function to a newline, use round brackets

BAD EXAMPLE:
```
const humans = people.map((a, b) =>
    a + b);

const babies = people.map(person => children 
 ? a
 : b
 
```

GOOD EXAMPLE: 
```
const humans = people.reduce((a, b) => (
  a + b
), 0);